From a373140bd4283ca0e5b4fc38c3762663d47e058a Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Tue, 7 Jun 2022 17:42:56 +0800
Subject: [PATCH] sw64: fix sendfile system call
Patch-mainline: NO, OTHERS
References: openEuler-22.03-LTS
Modified-by-SEL: No


Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I56XYC

--------------------------------

Using sys_sendfile will cause failure in sending large file(>=2GB).
This patch fixes the problem.

Signed-off-by: He Sheng <hesheng@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kernel/syscalls/syscall.tbl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/sw_64/kernel/syscalls/syscall.tbl b/arch/sw_64/kernel/syscalls/syscall.tbl
index e9dced7516ab..b9b93d70124d 100644
--- a/arch/sw_64/kernel/syscalls/syscall.tbl
+++ b/arch/sw_64/kernel/syscalls/syscall.tbl
@@ -377,7 +377,7 @@
 367	common	getcwd				sys_getcwd
 368	common	capget				sys_capget
 369	common	capset				sys_capset
-370	common	sendfile			sys_sendfile
+370	common	sendfile			sys_sendfile64
 371	common	setresgid			sys_setresgid
 372	common	getresgid			sys_getresgid
 373	common	dipc				sys_ni_syscall
-- 
2.34.1

